src/Components/Locations.js

```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect } from 'react';
+import axios from "axios";
+
+export default function Locations() {
+    const [list, setList] = useState([]);
+    const [showList, setShowList] = useState(false)
+    useEffect(() => {
+        generateList()
+    }, [])
+    const generateList = async () => {
+        try {
+            const res = await axios.get(
+                `https://ghibliapi.herokuapp.com/locations`
+            )
+            setList( res.data)
+        } catch (err) {
+            setList( "")
+        }
+    }
+    const handleClick = () => {
+        setShowList((prev) => !prev)
+    }
+    let listOfLocations = []
+    if (list.length) {
```

Comment:
I don't believe we need this condition here. If list doesn't have a length the map will just return an empty array.